src/Parser.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace CSVParser;
+
+class Parser {
+
+    var $path;
+    var $header_array = array();
+    var $delimiter;
+
+    function __construct ($path, $header_array = array(), $delimiter = ",") {
+
+        if(file_exists($path)){ //verify path
+            $this -> $path = $path; //fill path
+
+            if($delimiter == "," || $delimiter == "|" || $delimiter == "\t") { //verify delimiter
+                $this -> $delimiter = $delimiter; //fill delimiter
+
+                if(count($header_array) > 0) { //make sure header array is not empty
+                    //idea on foreach to fill array came from: stackoverflow.com/questions/3045619/how-to-store-values-from-foreach-loop-into-an-array
+                    foreach($header_array as $header) {
+                        $this -> $header_array[] = $header;
+                    }
+                } else {

```

Comment:
With this requests detailed above all of these `else's` will not be necessary. You'll find as I do code reviews that `else's`  are never actually needed. In rare cases I'll let an `else` slide if it makes the code easier to read.

Reply:
The else statements where because I didn't know how to throw the error that you had previously stated as needing.  It was my best idea at the time.